File: golang/docs/examples/browser-use/browser/visit_aliyun.go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay/browser"
	"github.com/playwright-community/playwright-go"
)

func main() {
	// Get API key from environment variable
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		log.Fatal("AGENTBAY_API_KEY environment variable not set")
	}

	// Initialize AgentBay client
	agentBay := agentbay.NewAgentBay(apiKey)

	// Create a session with browser image
	params := &agentbay.CreateSessionParams{
		ImageId: "browser_latest",
	}

	fmt.Println("Creating session...")
	sessionResult, err := agentBay.Create(params)
	if err != nil {
		log.Fatalf("Failed to create session: %v", err)
	}
	if !sessionResult.Success {
		log.Fatalf("Failed to create session: %s", sessionResult.ErrorMessage)
	}

	session := sessionResult.Session
	fmt.Printf("Session created: %s\n", session.GetSessionID())
	defer func() {
		fmt.Println("Deleting session...")
		session.Delete()
	}()

	// Initialize browser with default options
	fmt.Println("Initializing browser...")
	option := browser.NewBrowserOption()

	ok, err := session.Browser.Initialize(option)
	if err != nil {
		log.Fatalf("Browser initialization failed: %v", err)
	}
	if !ok {
		log.Fatal("Browser initialization returned false")
	}

	// Get CDP endpoint URL
	endpointURL, err := session.Browser.GetEndpointURL()
	if err != nil {
		log.Fatalf("Failed to get endpoint URL: %v", err)
	}
	fmt.Printf("Browser CDP endpoint: %s\n", endpointURL)

	// Connect Playwright over CDP
	fmt.Println("Starting Playwright...")
	pw, err := playwright.Run()
	if err != nil {
		log.Fatalf("Failed to start Playwright: %v", err)
	}
	defer pw.Stop()

	fmt.Println("Connecting to browser over CDP...")
	browserInstance, err := pw.Chromium.ConnectOverCDP(endpointURL)
	if err != nil {
		log.Fatalf("Failed to connect over CDP: %v", err)
	}
	defer browserInstance.Close()

	// Get existing browser context
	contexts := browserInstance.Contexts()
	if len(contexts) == 0 {
		log.Fatal("No browser contexts available")
	}
	context := contexts[0]

	// Create a new page
	page, err := context.NewPage()
	if err != nil {
		log.Fatalf("Failed to create new page: %v", err)
	}

	// Navigate to Aliyun
	fmt.Println("Navigating to https://www.aliyun.com...")
	_, err = page.Goto("https://www.aliyun.com")
	if err != nil {
		log.Fatalf("Failed to navigate: %v", err)
	}

	// Get and print page title
	title, err := page.Title()
	if err != nil {
		log.Fatalf("Failed to get title: %v", err)
	}
	fmt.Printf("Page title: %s\n", title)

	// Take a screenshot (optional)
	fmt.Println("Taking screenshot...")
	_, err = page.Screenshot(playwright.PageScreenshotOptions{
		Path: playwright.String("aliyun_screenshot.png"),
	})
	if err != nil {
		log.Printf("Warning: Failed to take screenshot: %v", err)
	} else {
		fmt.Println("Screenshot saved to aliyun_screenshot.png")
	}

	fmt.Println("Browser automation completed successfully!")
}

File: python/docs/examples/browser-use/browser/visit_aliyun.py
"""
Example demonstrating AIBrowser capabilites with AgentBay SDK.
This example shows how to use AIBrowser to visit aliyun.com, including:
- Create AIBrowser session
- Use playwright to connect to AIBrowser instance through CDP protocol
- Utilize playwright to visit aliyun.com
"""

import os
import time
import asyncio

from agentbay import AgentBay
from agentbay.session_params import CreateSessionParams
from agentbay.browser.browser import BrowserOption

from playwright.async_api import async_playwright

async def main():
    # Get API key from environment variable
    api_key = os.getenv("AGENTBAY_API_KEY")
    if not api_key:
        print("Error: AGENTBAY_API_KEY environment variable not set")
        return

    # Initialize AgentBay client
    print("Initializing AgentBay client...")
    agent_bay = AgentBay(api_key=api_key)

    # Create a session
    print("Creating a new session...")
    params = CreateSessionParams(
        image_id="browser_latest",  # Specify the image ID
    )
    session_result = agent_bay.create(params)

    if session_result.success:
        session = session_result.session
        print(f"Session created with ID: {session.session_id}")

        if await session.browser.initialize_async(BrowserOption()):
            print("Browser initialized successfully")
            endpoint_url = session.browser.get_endpoint_url()
            print("endpoint_url =", endpoint_url)

            async with async_playwright() as p:
                browser = await p.chromium.connect_over_cdp(endpoint_url)
                context = browser.contexts[0]
                page = await context.new_page()
                await page.goto("https://www.aliyun.com")
                print("page.title() =", await page.title())

                await page.wait_for_timeout(5000)

                # Modify page font to Microsoft YaHei
                await page.evaluate("""
                    document.body.style.fontFamily = 'Microsoft YaHei';
                """)

                await page.wait_for_timeout(5000)

                # Scale page content to 200%
                await page.evaluate("""
                    document.body.style.transform = 'scale(2)';
                    document.body.style.transformOrigin = 'top left';
                """)

                await page.wait_for_timeout(10000)
                await browser.close()
        agent_bay.delete(session)

if __name__ == "__main__":
    asyncio.run(main())
